src/utils/try.ts

```diff
@@ -0,0 +1,17 @@
+export class Try<T> {
```

Comment:
Not against this by any means – I've done a similar thing in our telemetry service – but would it be worth standardising on an approach here w/ e.g. [fp-ts](https://gcanti.github.io/fp-ts/modules/Either.ts.html), or do we think the overhead (comprehension, opinions) isn't worth paying?

Reply:
Wasn't entirely sure if `fp-ts` was too heavy right now, as it's for this single case; maybe adding `Try` to [`@guardian/types`](https://github.com/guardian/types) is an option?

Reply:
Definitely an `Option` – `Either` would be good (not necessarily for now!), I wouldn't want to be seen as `Invariant` (although I suspect weight doesn't matter much in a node environment?), interested in the medium term in the `Group` consensus

Reply:
I think @jamie-lynch 's [comment](https://github.com/guardian/cdk/pull/66/files#r545678705) is key here, understanding the use-case for a multi-app stack will tell us if we need this.